src/Microsoft.Health.Fhir.SqlServer/Features/ExceptionExtension.cs

```diff
@@ -21,8 +22,17 @@ internal static bool IsRetriable(this Exception e)
 
         internal static bool IsExecutionTimeout(this Exception e)
         {
-            var str = e.ToString().ToLowerInvariant();
-            return str.Contains("execution timeout expired", StringComparison.OrdinalIgnoreCase);
+            if (e is SqlException sqlException && sqlException.Number == -2)
```

Comment:
The main requirement in exception handling is that it should be as simple as possible. 
Logic added does not add value.